src/reachability/ai2.jl

```diff
@@ -55,34 +55,22 @@ function solve(solver::Ai2, problem::Problem)
     return check_inclusion(reach, problem.output)
 end
 
-forward_layer(solver::Ai2, L::Layer, inputs::Vector) = forward_layer.(solver, L, inputs)
-
-function forward_layer(solver::Ai2h, L::Layer{ReLU}, input::AbstractPolytope)
-    Ẑ = affine_map(L, input)
-    relued_subsets = forward_partition(L.activation, Ẑ) # defined in reachability.jl
-    return convex_hull(UnionSetArray(relued_subsets))
-end
-
-# method for Zonotope and Hyperrectangle, if the input set isn't a Zonotope
-function forward_layer(solver::Union{Ai2z, Box}, L::Layer{ReLU}, input::AbstractPolytope)
-    return forward_layer(solver, L, overapproximate(input, Hyperrectangle))
-end
-
-function forward_layer(solver::Ai2z, L::Layer{ReLU}, input::AbstractZonotope)
-    Ẑ = affine_map(L, input)
-    return overapproximate(Rectification(Ẑ), Zonotope)
-end
-
-
-function forward_layer(solver::Box, L::Layer{ReLU}, input::AbstractZonotope)
-    Ẑ = approximate_affine_map(L, input)
-    return rectify(Ẑ)
-end
-
-function forward_layer(solver::Ai2, L::Layer{Id}, input)
-    return affine_map(L, input)
-end
-
+# Ai2h and Ai2z use affine_map
+# Box uses approximate_affine_map for the linear region if it is propagating a zonotope
+forward_linear(solver::Union{Ai2h}, L::Layer{ReLU}, input::AbstractPolytope) = affine_map(L, input)
```

Comment:
Single union types generally don't make sense... Is there something you were trying to accomplish here?